buggy_calculator.py

```
@@ -24,50 +24,45 @@ def get_user_data(user_input):
 def recursive_factorial(n):
-    # Bug: missing base case, will cause infinite recursion
+    if n <= 1:
+        return 1
     return n * recursive_factorial(n - 1)
 
 
 def fetch_data(url):
-    # Bug: catching all exceptions silently
     try:
         import urllib.request
         response = urllib.request.urlopen(url)
```

Ruff (0.15.7)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)